web/client/epics/mediaEditor.js

```diff
@@ -309,3 +325,17 @@ export const setMediaTypeOnDisable = (action$) =>
             const mediaType = find(Object.values(MediaTypes), val => !includes(mediaTypes, val));
             return mediaType ? Observable.of(setMediaType(mediaType)) : Observable.empty();
         });
+
+/**
+ * Pairs media types with the last selected media service
+ * @memberof epics.mediaEditor
+ * @param {Observable} action$ stream of actions
+ */
+export const setMediaTypeSource = (action$) =>
```

Comment:
i'm not strongly against this solution, but i wanted to mention that this can be achieve also with a case in the mediaEditor reducer by editing SET_MEDIA_TYPE case

epics are side effect situations that must be used as last resort imho